ui/admin/app/controllers/scopes/scope/workers/index.js

```diff
@@ -65,17 +119,22 @@ export default class ScopesScopeWorkersIndexController extends Controller {
     return `${tag.key} = ${tag.value}`;
   }
 
+  // =actions
   /**
-   * Determine equality between two tags based on both key and value.
-   * @param {object} firstTag
-   * @param {object} secondTag
+   * Sets a query param to the value of selectedItems
+   * @param {string} paramKey
+   * @param {[string]} selectedItems
    */
-  isEqual(firstTag, secondTag) {
-    return firstTag.key === secondTag.key && firstTag.value === secondTag.value;
+  @action
+  applyFilter(paramKey, selectedItems) {
+    // Decoding the selectedItems
+    const decodedItems = selectedItems.map((item) => {
```

Comment:
Why do we decode back into an object instead of keeping it encoded for the query string and not use `encodedTags`? And just handle decoding back in the model hook

Reply:
I decoded it here so the query strings would show human-readable filters.  But if we are ok with having encoded values in the URL, I can move the decoding to the model hook instead

Reply:
Yeah I think it's best to have it in the URL encoded since it's not great to have literal objects in the query strings